network-config-analyzer/MinDFA.py

```diff
@@ -0,0 +1,269 @@
+        return res
+
+    @staticmethod
+    def dfa_all_words(legal_characters):
+        """
+        get MinDFA for all words in a domain
+        :param legal_characters: str object with regular expression for all words in a domain
+        :return: MinDFA object such that its language is equivalent to all words in the domain
+        """
+        res = MinDFA.dfa_from_regex(legal_characters)
+        res.is_all_words = MinDFA.Ternary.TRUE
+        return res
+
+    # TODO: this function may not be necessary, if keeping the current __eq__ override
+    def is_dfa_wll_words(self, all_words_dfa):
+        """
+        return True iff self is equivalent to DFA of all words.
+        avoid dfa-comparison if possible (rely on is_all_words and len() when possible )
+        :param all_words_dfa: the DFA of all words for the relevant domain.
+        :rtype: bool
+        """
+        if self.is_all_words == MinDFA.Ternary.TRUE:
+            return True
+        elif self.is_all_words == MinDFA.Ternary.UNKNOWN and not self.has_finite_len() and self == all_words_dfa:
+            return True
+        return False
+
+    def copy(self):
+        res = MinDFA.dfa_from_fsm(self)
+        res.is_all_words = self.is_all_words
+        if self.complement_dfa is not None:
+            res.complement_dfa = self.complement_dfa.copy()
+        return res
+
+    def __hash__(self):
+        return hash((frozenset(self.states), frozenset(self.finals), frozenset(self.map), self.initial))
+        # return hash((frozenset(self.alphabet), frozenset(self.states), frozenset(self.finals), frozenset(self.map), self.initial ))
+
+    def _get_strings_set_str(self):
+        """
+        This method assumes that self has a finite len.
+        Returns a set of strings with all str values in the language of self.
+        :rtype: set of strings
+        """
+        str_values = set()
+        str_generator = self.strings()
+        for i in range(0, len(self)):
+            str_val = next(str_generator)
+            str_val_new = ''.join(ch for ch in str_val)
+            str_values |= {str_val_new}
+        return str(str_values)
+
+    def has_finite_len(self):
+        """
+        :return: True iff self has finite number of words accepted by it.
+        :rtype: bool
+        """
+        if self.is_all_words == MinDFA.Ternary.TRUE:
+            return False
+        try:
+            len(self)
+            return True
+        except OverflowError:
+            return False
+
+    def __str__(self):
+        """
+        str representation of the language accepted by this DFA:
+        - option 1: if language has finite number of words -> return string with all accepted words.
+        - option 2 (costly): convert fsm to regex with greenery
+        :rtype: str
+        """
+        if self.has_finite_len():
+            return self._get_strings_set_str()
+        # TODO: consider performance implications of this conversion from MinDFA to regex
```

Comment:
Maybe also check `if self.is_all_words == MinDFA.Ternary.TRUE` ?
If so (and the language is infinite) we can simply print '*'.

Reply:
done